controllers/release/adapter.go

```diff
@@ -769,6 +1115,27 @@ func (a *adapter) registerManagedProcessingData(releasePipelineRun *tektonv1.Pip
 	return a.client.Status().Patch(a.ctx, a.release, patch)
 }
 
+// registerTenantCollectorsProcessingStatus updates the status of the Release being processed by monitoring the status of the
+// associated tenant collectors Release PipelineRun and setting the appropriate state in the Release. If the PipelineRun hasn't
+// started/succeeded, no action will be taken.
+func (a *adapter) registerTenantCollectorsProcessingStatus(pipelineRun *tektonv1.PipelineRun) error {
+	if pipelineRun == nil || !pipelineRun.IsDone() {
+		return nil
+	}
+
+	patch := client.MergeFrom(a.release.DeepCopy())
+
+	condition := pipelineRun.Status.GetCondition(apis.ConditionSucceeded)
+	if condition.IsTrue() {
+		a.release.MarkTenantCollectorsPipelineProcessed()
+	} else {
+		a.release.MarkTenantCollectorsPipelineProcessingFailed(condition.Message)
+		a.release.MarkReleaseFailed("Release processing failed on tenant collectors pipelineRun")
+	}
+
+	return a.client.Status().Patch(a.ctx, a.release, patch)
+}
+
 // registerTenantProcessingStatus updates the status of the Release being processed by monitoring the status of the
```

Comment:
Bit lost with this comment. Could you clarify?

Reply:
If a phase fails, we set the Release as failed. However, only in registerTenantProcessingStatus, we also set ManagedProcessing as skipped

Reply:
It seems that it comes from the time we allowed to [run both tenant and managed pipelines at the same time](https://github.com/konflux-ci/release-service/commit/9d2ba529e8d9b24e3e5c2fc2737da27a30ce5ada#diff-6f40aa8049566b39f0c38b10d839c98d6b9298c98d783c4b82d8e6ba368067d4).

I think this could be the mechanism used to avoid the Managed pipeline to run if the tenant failed. Without it, as EnsureManagedPipelineIsProcessed comes after the status is set but the status doesn't requeue unlesss there's an error, the managed pipelineRun would get created (the gates we use ignore the results of previous phases) and we don't want this to happen.

I removed that line and changed to gate to:
```
if a.release.HasManagedPipelineProcessingFinished() || !a.release.HasTenantPipelineProcessingFinished() ||
		!a.release.IsTenantPipelineProcessed() {
```

I think this is ok as we don't want the managed pipeline to run if the tenant failed. I did the same for the managed collectors pipeline and added tests.

Please take a look and verify. I could be wrong.